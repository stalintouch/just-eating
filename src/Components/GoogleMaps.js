/* global google */
import React, { Component } from 'react';

// this component creates a google map with the received props
export default class GoogleMaps extends Component {
  
  componentDidMount() {
    const {lat, lng} = this.props
    var map = new google.maps.Map(this.refs.map, {
      zoom: 12,
      center: {
        lat,
        lng
      }
    });
    var myCenter = new google.maps.LatLng(lat, lng);

    var marker = new google.maps.Marker({
      position: myCenter,
    });
    marker.setMap(map)
  }

  render() {
    return <div ref="map" style={{ height: '100%', width: '100%' }} />;
  }
}
