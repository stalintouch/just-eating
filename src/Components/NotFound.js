import React from 'react';

// functional component to have it render when no restaurants are found
const NotFound = () =>{
  return(
  <div>
    <h2 className="not-found">Not Restaurants found, please try searching for another city</h2>
  </div>
  )
}

export default NotFound;