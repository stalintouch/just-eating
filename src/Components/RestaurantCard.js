import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import Toggle from 'material-ui/Toggle';
import GoogleMaps from './GoogleMaps';
import {
  Card,
  CardActions,
  CardHeader,
  CardMedia,
  CardTitle,
  CardText
} from 'material-ui/Card';

export default class RestaurantCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false
    };
  }

  handleExpandChange = expanded => {
    this.setState({ expanded: expanded });
  };

  handleToggle = (event, toggle) => {
    this.setState({ expanded: toggle });
  };

  handleExpand = () => {
    this.setState({ expanded: true });
  };

  handleReduce = () => {
    this.setState({ expanded: false });
  };

  render() {
    const cardStyle = {
      width: '100%',
      background: `linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), 
              url(${this.props.backgroundImage}) no-repeat no-repeat center`,
      minHeight: '200px',
      marginBottom: '20px',
      display: 'flex',
      alignItems: 'center',
      backgroundSize: 'cover',
      color: '#fff',
      position: 'relative'
    };

    const {
      name,
      address,
      city,
      area,
      postal_code,
      country,
      price,
      phone,
      reserve_url,
      image_url,
      lat,
      lng
    } = this.props.restaurantInfo;

    return (
      <Card
        style={cardStyle}
        expanded={this.state.expanded}
        onExpandChange={this.handleExpandChange}
      >
        <CardHeader
          titleStyle={{ fontSize: '20px' }}
          titleColor="#fff"
          subtitleColor="#fff"
          title={name}
          subtitle={address}
          avatar={image_url}
          actAsExpander={true}
          showExpandableButton={false}
        />

        <CardTitle
          title={`Serving ${area}`}
          subtitle={`Price: ${price}`}
          expandable={true}
          titleColor="#fff"
          subtitleColor="#fff"
        />

        <CardText expandable={true} color="#fff">
          <div className="googlemaps">
            <GoogleMaps lat={lat} lng={lng} />
          </div>
          <ul className="card-ul">
            <li>
              Full Address: {address}, {postal_code}{' '}
            </li>
            <li>Phone: {phone}</li>
            <li>
              Website:{' '}
              <a
                href={reserve_url}
                title={`${name} restaurant`}
                target="_blank"
              >
                {reserve_url}
              </a>
            </li>
            <li />
          </ul>
        </CardText>

        <CardActions>
          <RaisedButton
            label="Read more"
            primary={true}
            onClick={this.handleExpand}
          />
          <RaisedButton
            label="Read less"
            secondary={true}
            onClick={this.handleReduce}
          />
        </CardActions>
      </Card>
    );
  }
}
