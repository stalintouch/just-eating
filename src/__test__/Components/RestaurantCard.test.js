import RestaurantCard from '../../Components/RestaurantCard';
import GoogleMaps from '../../Components/GoogleMaps';
import NotFound from '../../Components/NotFound';

describe('<RestaurantCard />', () => {
  let restaurantInfo;
  let wrapper;
  beforeEach(() => {
    restaurantInfo = {
      name: 'test',
      address: 'test',
      area: 'test',
      postal_code: 'test',
      price: 1,
      phone: '555-555-5555',
      reserve_url: 'test',
      image_url: 'test',
      lat: 'test',
      lng: 'test'
    };
    wrapper = shallow(<RestaurantCard {...{ restaurantInfo }} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('render a google map component without issues', () => {
    expect(wrapper.find(GoogleMaps)).toHaveLength(1);
  });
});
