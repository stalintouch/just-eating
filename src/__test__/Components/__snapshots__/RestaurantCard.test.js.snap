// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestaurantCard /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantCard
    restaurantInfo={
        Object {
            "address": "test",
            "area": "test",
            "image_url": "test",
            "lat": "test",
            "lng": "test",
            "name": "test",
            "phone": "555-555-5555",
            "postal_code": "test",
            "price": 1,
            "reserve_url": "test",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CardHeader
          actAsExpander={true}
          avatar="test"
          showExpandableButton={false}
          subtitle="test"
          subtitleColor="#fff"
          title="test"
          titleColor="#fff"
          titleStyle={
                    Object {
                              "fontSize": "20px",
                            }
          }
/>,
        <CardTitle
          expandable={true}
          subtitle="Price: 1"
          subtitleColor="#fff"
          title="Serving test"
          titleColor="#fff"
/>,
        <CardText
          color="#fff"
          expandable={true}
>
          <ul
                    className="card-ul"
          >
                    <li>
                              Full Address: 
                              test
                              , 
                              test
                    </li>
                    <li>
                              Phone: 
                              555-555-5555
                    </li>
                    <li>
                              Website: 
                              <a
                                        href="test"
                                        target="_blank"
                                        title="test restaurant"
                              >
                                        test
                              </a>
                    </li>
                    <li />
          </ul>
          <div
                    className="googlemaps"
          >
                    <GoogleMaps
                              lat="test"
                              lng="test"
                    />
          </div>
</CardText>,
        <CardActions>
          <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="Read more"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={true}
                    secondary={false}
          />
</CardActions>,
      ],
      "expandable": false,
      "expanded": false,
      "initiallyExpanded": false,
      "onExpandChange": [Function],
      "style": Object {
        "alignItems": "center",
        "background": "linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), 
              url(undefined) no-repeat no-repeat center",
        "backgroundSize": "cover",
        "color": "#fff",
        "display": "flex",
        "marginBottom": "20px",
        "minHeight": "200px",
        "position": "relative",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actAsExpander": true,
          "avatar": "test",
          "showExpandableButton": false,
          "subtitle": "test",
          "subtitleColor": "#fff",
          "title": "test",
          "titleColor": "#fff",
          "titleStyle": Object {
            "fontSize": "20px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "expandable": true,
          "subtitle": "Price: 1",
          "subtitleColor": "#fff",
          "title": "Serving test",
          "titleColor": "#fff",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ul
              className="card-ul"
>
              <li>
                            Full Address: 
                            test
                            , 
                            test
              </li>
              <li>
                            Phone: 
                            555-555-5555
              </li>
              <li>
                            Website: 
                            <a
                                          href="test"
                                          target="_blank"
                                          title="test restaurant"
                            >
                                          test
                            </a>
              </li>
              <li />
</ul>,
            <div
              className="googlemaps"
>
              <GoogleMaps
                            lat="test"
                            lng="test"
              />
</div>,
          ],
          "color": "#fff",
          "expandable": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Full Address: 
                  test
                  , 
                  test
</li>,
                <li>
                  Phone: 
                  555-555-5555
</li>,
                <li>
                  Website: 
                  <a
                                    href="test"
                                    target="_blank"
                                    title="test restaurant"
                  >
                                    test
                  </a>
</li>,
                <li />,
              ],
              "className": "card-ul",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Full Address: ",
                    "test",
                    ", ",
                    "test",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Full Address: ",
                  "test",
                  ", ",
                  "test",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Phone: ",
                    "555-555-5555",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Phone: ",
                  "555-555-5555",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Website: ",
                    <a
                      href="test"
                      target="_blank"
                      title="test restaurant"
>
                      test
</a>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Website: ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "test",
                      "href": "test",
                      "target": "_blank",
                      "title": "test restaurant",
                    },
                    "ref": null,
                    "rendered": "test",
                    "type": "a",
                  },
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <GoogleMaps
                lat="test"
                lng="test"
/>,
              "className": "googlemaps",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "lat": "test",
                "lng": "test",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Read more"
              labelPosition="after"
              onClick={[Function]}
              primary={true}
              secondary={false}
/>,
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "fullWidth": false,
              "label": "Read more",
              "labelPosition": "after",
              "onClick": [Function],
              "primary": true,
              "secondary": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CardHeader
            actAsExpander={true}
            avatar="test"
            showExpandableButton={false}
            subtitle="test"
            subtitleColor="#fff"
            title="test"
            titleColor="#fff"
            titleStyle={
                        Object {
                                    "fontSize": "20px",
                                  }
            }
/>,
          <CardTitle
            expandable={true}
            subtitle="Price: 1"
            subtitleColor="#fff"
            title="Serving test"
            titleColor="#fff"
/>,
          <CardText
            color="#fff"
            expandable={true}
>
            <ul
                        className="card-ul"
            >
                        <li>
                                    Full Address: 
                                    test
                                    , 
                                    test
                        </li>
                        <li>
                                    Phone: 
                                    555-555-5555
                        </li>
                        <li>
                                    Website: 
                                    <a
                                                href="test"
                                                target="_blank"
                                                title="test restaurant"
                                    >
                                                test
                                    </a>
                        </li>
                        <li />
            </ul>
            <div
                        className="googlemaps"
            >
                        <GoogleMaps
                                    lat="test"
                                    lng="test"
                        />
            </div>
</CardText>,
          <CardActions>
            <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Read more"
                        labelPosition="after"
                        onClick={[Function]}
                        primary={true}
                        secondary={false}
            />
</CardActions>,
        ],
        "expandable": false,
        "expanded": false,
        "initiallyExpanded": false,
        "onExpandChange": [Function],
        "style": Object {
          "alignItems": "center",
          "background": "linear-gradient( rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), 
              url(undefined) no-repeat no-repeat center",
          "backgroundSize": "cover",
          "color": "#fff",
          "display": "flex",
          "marginBottom": "20px",
          "minHeight": "200px",
          "position": "relative",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actAsExpander": true,
            "avatar": "test",
            "showExpandableButton": false,
            "subtitle": "test",
            "subtitleColor": "#fff",
            "title": "test",
            "titleColor": "#fff",
            "titleStyle": Object {
              "fontSize": "20px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "expandable": true,
            "subtitle": "Price: 1",
            "subtitleColor": "#fff",
            "title": "Serving test",
            "titleColor": "#fff",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <ul
                className="card-ul"
>
                <li>
                                Full Address: 
                                test
                                , 
                                test
                </li>
                <li>
                                Phone: 
                                555-555-5555
                </li>
                <li>
                                Website: 
                                <a
                                                href="test"
                                                target="_blank"
                                                title="test restaurant"
                                >
                                                test
                                </a>
                </li>
                <li />
</ul>,
              <div
                className="googlemaps"
>
                <GoogleMaps
                                lat="test"
                                lng="test"
                />
</div>,
            ],
            "color": "#fff",
            "expandable": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    Full Address: 
                    test
                    , 
                    test
</li>,
                  <li>
                    Phone: 
                    555-555-5555
</li>,
                  <li>
                    Website: 
                    <a
                                        href="test"
                                        target="_blank"
                                        title="test restaurant"
                    >
                                        test
                    </a>
</li>,
                  <li />,
                ],
                "className": "card-ul",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Full Address: ",
                      "test",
                      ", ",
                      "test",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Full Address: ",
                    "test",
                    ", ",
                    "test",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Phone: ",
                      "555-555-5555",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Phone: ",
                    "555-555-5555",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Website: ",
                      <a
                        href="test"
                        target="_blank"
                        title="test restaurant"
>
                        test
</a>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Website: ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "test",
                        "href": "test",
                        "target": "_blank",
                        "title": "test restaurant",
                      },
                      "ref": null,
                      "rendered": "test",
                      "type": "a",
                    },
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <GoogleMaps
                  lat="test"
                  lng="test"
/>,
                "className": "googlemaps",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "lat": "test",
                  "lng": "test",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Read more"
                labelPosition="after"
                onClick={[Function]}
                primary={true}
                secondary={false}
/>,
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Read more",
                "labelPosition": "after",
                "onClick": [Function],
                "primary": true,
                "secondary": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
